test/e2e/suite/mixed_mode/mixed_mode_test.go

```diff
@@ -0,0 +1,408 @@
+)
+
+func init() {
+	flag.StringVar(&filePath, "filepath", "", "Path to configuration")
+}
+
+func TestMixedModeE2E(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Mixed Mode E2E Suite")
+}
+
+var _ = SynchronizedBeforeSuite(
+	// This function only runs once, on the first process
+	func(ctx context.Context) []byte {
+		suiteConfig := suite.BeforeSuiteCredentialSetup(ctx, filePath)
+
+		test := suite.BeforeVPCTest(ctx, &suiteConfig)
+		credentialProviders := suite.AddClientsToCredentialProviders(suite.CredentialProviders(), test)
+		osProviderList := suite.OSProviderList(credentialProviders)
+		randomOSProvider := osProviderList[rand.Intn(len(osProviderList))]
+
+		hybridNode := suite.NodeCreate{
+			InstanceName: "mixed-mode-hybrid",
+			InstanceSize: e2e.Large,
+			NodeName:     "mixed-mode-hybrid",
+			OS:           randomOSProvider.OS,
+			Provider:     randomOSProvider.Provider,
+			ComputeType:  e2e.CPUInstance,
+		}
+		suite.CreateNodes(ctx, test, []suite.NodeCreate{hybridNode})
+
+		Expect(test.CreateManagedNodeGroups(ctx)).To(Succeed(), "managed node group should be created successfully")
+
+		// Ensure mixed mode connectivity by adding required security group rules
+		err := ensureMixedModeConnectivity(ctx, test)
+		Expect(err).NotTo(HaveOccurred(), "Mixed mode connectivity rules should be added successfully")
+
+		suiteJson, err := yaml.Marshal(suiteConfig)
+		Expect(err).NotTo(HaveOccurred(), "suite config should be marshalled successfully")
+		return suiteJson
+	},
+	// This function runs on all processes
+	func(ctx context.Context, data []byte) {
+		// add a small sleep to add jitter to the start of each test
+		randomSleep := rand.Intn(10)
+		err := smithyTime.SleepWithContext(ctx, time.Duration(randomSleep)*time.Second)
+		Expect(err).NotTo(HaveOccurred(), "failed to sleep")
+		suiteConfig = suite.BeforeSuiteCredentialUnmarshal(ctx, data)
+	},
+)
+
+var _ = Describe("Mixed Mode Testing", func() {
+	When("hybrid and cloud-managed nodes coexist", func() {
+		var test *suite.PeeredVPCTest
+
+		BeforeEach(func(ctx context.Context) {
+			test = suite.BeforeVPCTest(ctx, suiteConfig)
+
+			// Comprehensive cleanup before each test
+			test.Logger.Info("Running comprehensive cleanup to ensure clean state")
+			cleanupTestResources(ctx, test)
+		})
+
+		AfterEach(func(ctx context.Context) {
+			test.Logger.Info("Running comprehensive cleanup after test")
+			cleanupTestResources(ctx, test)
+		})
+
+		Context("Pod-to-Pod Communication", func() {
+			It("should enable cross-VPC pod-to-pod communication from hybrid to cloud nodes", func(ctx context.Context) {
+				cloudPod := &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "nginx-cloud",
+						Namespace: "default",
+						Labels:    map[string]string{"app": "nginx-cloud"},
+					},
+					Spec: corev1.PodSpec{
+						NodeSelector: cloudNodeSelector,
+						Containers: []corev1.Container{{
+							Name:    "nginx-cloud",
+							Image:   "nginx:1.21",
```

Comment:
Can we use the nginx image used in the nginx pod helper method [here](https://github.com/aws/eks-hybrid/blob/main/test/e2e/kubernetes/pod.go#L44)? This helps to avoid the Dockerhub dependency and unauthenticated pull rate limits. Also if you can find a way to reuse that method instead of creating a new pod object from scratch, that would be great.

Also the 1.21 version seems to be pretty old, ~3 years. Why are we using that version? Can we bump to newer versions? Or as suggested above, you could use the image from `CreateNginxPodInNode` method which gives you the latest one.